code/range/ViewModel/src/main/java/ru/unn/agile/range/viewmodel/ViewModel.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    public BooleanProperty isContainsButtonDisabled() {
+        return btnContainsDisabled;
+    }
+
+    public BooleanProperty isEqualsButtonDisabled() {
+        return btnEqualsDisabled;
+    }
+
+    public BooleanProperty isOverlapsButtonDisabled() {
+        return btnOverlapsDisabled;
+    }
+
+    public StringProperty getTxtRange() {
+        return txtRange;
+    }
+
+    public StringProperty getTxtResult() {
+        return txtResult;
+    }
+
+    public StringProperty getTxtInput() {
+        return txtInput;
+    }
+
+    public void containsInput() {
+        String input = txtInput.get();
+
+        if (isInteger(input)) {
+            if (range.containsValue(Integer.parseInt(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+        if (isIntegerSet(input)) {
+            if (range.containsSet(Arrays.stream(input.substring(1, input.length() - 1).split(","))
+                    .map(String::trim).mapToInt(Integer::parseInt).toArray())) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+        if (isRange(input)) {
+            if (range.containsRange(new Range(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+    }
+
+    public void overlapsRange() {
+        String input = txtInput.get();
+        if (isRange(input)) {
+            if (range.overlapsRange(new Range(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+    }
+
+    public void equalsRange() {
+        String input = txtInput.get();
+        if (isRange(input)) {
+            if (range.equals(new Range(input))) {
+                txtResult.setValue("Yes");
+            } else {
+                txtResult.setValue("No");
+            }
+        }
+    }
+
+    public void getAllPoints() {
+        txtResult.setValue(Arrays.toString(range.getAllPoints()));
+    }
+
+    public void getEndPoints() {
+        txtResult.setValue(Arrays.toString(range.endPoints()));
+    }
+
+
+    private void setRange(final String input) {
+
+        var isCorrectInput = true;
+
+        try {
+            this.range = new Range(input);
+        } catch (IllegalArgumentException ex) {
+            isCorrectInput = false;
+        }
+
+        btnGetAllPointsDisabled.setValue(!isCorrectInput);
+        btnGetEndPointsDisabled.setValue(!isCorrectInput);
+        disableAllButtonsConnectedWithInput();
+    }
+
+    private void setInput(final String input) {
+        disableAllButtonsConnectedWithInput();
+
+        if (btnGetEndPointsDisabled.get()) {
+            return;
+        }
+
+        if (isInteger(input) || isIntegerSet(input)) {
+            btnContainsDisabled.setValue(false);
```

Comment:
In this place you can declare a boolean variable "isDisabled"

Reply:
fixed